### PR TITLE
feat(cli/repl) add tab completion

### PR DESCRIPTION
**Pull request by caspervonb**
This adds tab completion for global identifiers and object properties.

Closes https://github.com/denoland/deno/issues/3086

### CODE REVIEW

File: cli/repl.rs [hunk 1]
```diff
@@ -59,17 +203,27 @@ async fn post_message_and_poll(
...
+  response_tx: &Sender<Result<Value, AnyError>>,
   editor: Arc<Mutex<Editor<Helper>>>,
 ) -> Result<String, ReadlineError> {
   let mut line =
     tokio::task::spawn_blocking(move || editor.lock().unwrap().readline("> "));
 
   let mut poll_worker = true;
+
   loop {
+    for (method, params) in message_rx.try_iter() {
```
Comment (caspervonb) [inline cli/repl.rs:217]: Make this a stream and select on it below.

**Comment by caspervonb** [comment]
Alright so right the first pass only supports a few simple scenarios single line scenarios

```
Deno.|<tab>
Deno.write|<tab>
Deno[|<tab>
Deno["write|<tab>
```

Next step would be for completions for multi-line "lines" based on the current cursor position

```
const foo = Deno.|<tab>
(
  Deno.write<tab>
)
false, Deno[<tab>
true && Deno["write<tab>
```

And also word completions

```
De|<tab>
let foo = win|<tab>
(
  con|<tab>
)
```

**Comment by ry** [comment]
Tested it locally - seems to work! Very nice.

It only works for `Deno.|<tab>` tho. When I do `windo|<tab>` it doesn't complete to "window".

**Comment by caspervonb** [comment]
> . When I do windo|<tab> it doesn't complete to "window".

Yet to be implemented, only the rhs of propery accessors are completed at the moment 🙃

**Review by crypt096 — APPROVED** [review]

**Comment by caspervonb** [comment]
Slightly confusing, but line in a completer is actually a line. Whereas a line in a highlighter is all lines... so multi-line completion like the example below does not work at the moment

```
(
window
).|<tab>
```

**Comment by bartlomieju** [comment]
@caspervonb please rebase

**Comment by caspervonb** [comment]
Will do, merge #7896 first please :)

**Review by bartlomieju — APPROVED** [review]
LGTM, very cool feature 🚀 

In future we should consider changing colors a bit for autocompleted items or even displaying list of all possible suggestions.

**Comment by caspervonb** [comment]
There are other completion modes available, specifically there's a list list-mode that interoperates with "hints" (the one trait we haven't touched on yet in rustyline).

So yeah that's open ended and should be relatively accessible for tweaking.

**Pull request merged**